src/query_plan/operation.rs

```diff
@@ -747,6 +809,43 @@ pub(crate) mod normalized_field_selection {
         }
     }
 
+    impl NormalizedFieldSelection {
+        pub(crate) fn with_updated_selection_set(
+            &self,
+            selection_set: Option<NormalizedSelectionSet>,
+        ) -> Self {
+            Self {
+                field: self.field.clone(),
+                selection_set,
+            }
+        }
+
+        pub(crate) fn element(&self) -> OpPathElement {
+            OpPathElement::Field(self.field.clone())
+        }
+
+        pub(crate) fn with_updated_alias(&self, alias: Name) -> NormalizedField {
+            let mut data = self.field.data().clone();
+            data.alias = Some(alias);
+            NormalizedField::new(data)
+            // FIXME: what is copyAttachments?
```

Comment:
"attachments" in JS is `sibling_typename` in rust.

Reply:
Alright, so it’s already part of `NormalizedFieldData` and nothing else is needed here.